src/Hint/Fixities.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE ViewPatterns, ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-
+
+Raise a warning if you have redundant brackets in nested infix expressions.
+
+<TEST>
+yes = 1 + (2 * 3) -- @Ignore 1 + 2 * 3
+no = (1 + 2) * 3
+no = 1 + 2 * 3
+</TEST>
+-}
+
+module Hint.Fixities(fixitiesHint) where
+
+import Hint.Type(DeclHint,Idea(..),rawIdea,Severity(..),toSS)
+import Config.Type
+import Data.Data(Data)
+import Data.List.Extra
+import Data.Map(Map, singleton, fromList)
+import Data.Generics.Uniplate.DataOnly
+import Fixity
+import Refact.Types
+
+import GHC.Hs
+import Outputable
+import SrcLoc
+import GHC.Util
+import Language.Haskell.GhclibParserEx.GHC.Utils.Outputable
+
+fixitiesHint :: [Setting] -> DeclHint
+fixitiesHint settings _ _ x =
+  concatMap (infixBracket fixities) (childrenBi x :: [LHsExpr GhcPs]) ++
+  concatMap (infixBracket fixities) (childrenBi x :: [LHsType GhcPs]) ++
```

Comment:
Why bother doing this given we know fixity hints only apply to expressions?

Reply:
Done